test/multiple-user-principal-types.test.js

```diff
@@ -373,17 +382,46 @@ describe('Multiple users with custom principalType', function() {
 
         return Album.create({name: 'album', userId: userFromOneModel.id})
           .then(function(album) {
-            return Role.isInRole(
-              Role.OWNER,
-              {
-                principalType: OneUser.modelName,
-                principalId: userFromOneModel.id,
-                model: Album,
-                id: album.id,
-              });
+            var validContext = {
+              principalType: OneUser.modelName,
+              principalId: userFromOneModel.id,
+              model: Album,
+              id: album.id,
+            };
+            return Role.isInRole(Role.OWNER, validContext);
           })
-          .then(function(isInRole) {
-            expect(isInRole).to.be.true();
+          .then(function(isOwner) {
+            expect(isOwner).to.be.true();
+          });
+      });
+
+      it('expects OWNER to resolve false if owner has incorrect principalType', function() {
+        var Album = app.registry.createModel('Album', {
+          name: String,
+          userId: Number,
+        }, {
+          relations: {
+            user: {
+              type: 'belongsTo',
+              model: 'OneUser',
+              foreignKey: 'userId',
+            },
+          },
+        });
+        app.model(Album, {dataSource: 'db'});
+
+        return Album.create({name: 'album', userId: userFromOneModel.id})
+          .then(function(album) {
+            var invalidContext = {
+              principalType: AnotherUser.modelName,
+              principalId: userFromOneModel.id,
+              model: Album,
+              id: album.id,
+            };
+            return Role.isInRole(Role.OWNER, invalidContext);
+          })
+          .then(function(isOwner) {
+            expect(isOwner).to.be.false();
           });
```

Comment:
split the OWNER test in 2 to check apart if resolver returns false when principalType is incorrect